zerver/views/previewable.py

```diff
@@ -0,0 +1,34 @@
+import re
+from urllib.parse import urlparse
+
+from django.http import HttpRequest, HttpResponse
+from django.utils.translation import gettext as _
+
+from zerver.lib.exceptions import JsonableError
+from zerver.lib.github import get_issue_or_pr_data
+from zerver.lib.request import REQ, has_request_variables
+from zerver.lib.response import json_success
+from zerver.models import UserProfile
+
+
+@has_request_variables
+def get_previewable_data(
+    request: HttpRequest, user_profile: UserProfile, url: str = REQ()
+) -> HttpResponse:
+    parsed_url = urlparse(url)
```

Comment:
`urlparse` can fail (e.g. with `https://boom[here`), which we should catch gracefully rather than 500.

Reply:
I went through codebase, we are not returning 422(Unprocessable Content) status code anywhere. Should we consider this for these type of errors?